docs/java-rest/high-level/ml/upgrade-job-model-snapshot.asciidoc

```diff
@@ -0,0 +1,42 @@
+--
+:api: upgrade-job-model-snapshot
+:request: UpgradeJobModelSnapshotRequest
+:response: UpgradeJobModelSnapshotResponse
+--
+[role="xpack"]
+
+[id="{upid}-{api}"]
+=== Upgrade job snapshot API
+
+Upgrades a previously stored {ml} model snapshot to the
+current major version.
+It accepts an +{request}+ object and responds
+with an +{response}+ object.
+
+[id="{upid}-{api}-request"]
+==== Upgrade job snapshots request
+
+A +{request}+ requires the following arguments:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request]
+--------------------------------------------------
+<1> The job that owns the snapshot
+<2> The snapshot id to upgrade
+<3> The time out of the request
```

Comment:
As an end user I would be interested in what this means if `wait_for_completion=false`, and what the default is.